.ci/end2end.groovy

```diff
@@ -37,93 +41,195 @@ pipeline {
+                      archiveArtifacts(allowEmptyArchive: true, artifacts: 'apm-its-docker.log')
+                    }
+                    archiveArtifacts(allowEmptyArchive: true, artifacts: 'tmp/*.log')
+                  }
+                }
+                unsuccessful {
+                  notifyTestStatus(env.GITHUB_CHECK_APM_UI, 'Test failures', 'FAILURE')
+                }
+                success {
+                  notifyTestStatus(env.GITHUB_CHECK_APM_UI, 'Tests passed', 'SUCCESS')
+                }
+              }
+            }
+          }
+          post {
+            always {
+              dir("${BASE_DIR}"){
+                archiveArtifacts(allowEmptyArchive: true, artifacts: "${E2E_DIR}/kibana.log")
+              }
             }
-            archiveArtifacts(allowEmptyArchive: true, artifacts: 'tmp/*.log')
           }
         }
-        unsuccessful {
-          notifyTestStatus('Test failures', 'FAILURE')
-        }
-        success {
-          notifyTestStatus('Tests passed', 'SUCCESS')
+        stage('UPTIME-UI') {
+          // This stage requires a new agent
+          agent { label 'linux && immutable' }
+          options { skipDefaultCheckout() }
+          when { expression { return env.RUN_UPTIME_E2E != "false" } }
+          stages {
+            stage('Prepare build context') {
+              options { skipDefaultCheckout() }
+              steps {
+                pipelineManager([ cancelPreviousRunningBuilds: [ when: 'PR' ] ])
+                deleteDir()
+                gitCheckout(basedir: "${BASE_DIR}", githubNotifyFirstTimeContributor: false,
+                            shallow: false, reference: "/var/lib/jenkins/.git-references/kibana.git")
+              }
+            }
+            stage('Prepare Kibana') {
+              options { skipDefaultCheckout() }
+              environment {
+                JENKINS_NODE_COOKIE = 'dontKillMe'
+              }
+              steps {
+                notifyStatus(env.GITHUB_CHECK_UPTIME_UI, 'Preparing kibana', 'PENDING')
+                dir("${BASE_DIR}"){
+                  sh "${UPTIME_E2E_DIR}/ci/prepare_kibana.sh"
+                }
+              }
+              post {
+                unsuccessful {
+                  notifyStatus(env.GITHUB_CHECK_UPTIME_UI, 'Kibana warm up failed', 'FAILURE')
+                }
+              }
+            }
+            stage('Smoke Tests'){
+              options { skipDefaultCheckout() }
+              steps{
+                notifyTestStatus(env.GITHUB_CHECK_UPTIME_UI, 'Running smoke tests', 'PENDING')
+                dir("${BASE_DIR}"){
+                  sh "${UPTIME_E2E_DIR}/ci/run_e2e.sh"
+                }
+              }
+              post {
+                always {
+                  dir("${UPTIME_E2E_DIR}"){
+                    sh 'docker-compose logs > uptime-e2e-docker.log || true'
+                    archiveArtifacts(allowEmptyArchive: true, artifacts: 'uptime-e2e-docker.log')
```

Comment:
Once L159 produces the junit files, then you can add something like

```suggestion
                    archiveArtifacts(allowEmptyArchive: true, artifacts: 'uptime-e2e-docker.log')
                    junit(allowEmptyResults: true, testResults: '<RELATIVE_PATH_FROM_UPTIME_E2E_DIR>/<FILE_NAME>.xml')
```

But please replace  `'<RELATIVE_PATH_FROM_UPTIME_E2E_DIR>/<FILE_NAME>.xml'`